socket.io/socket.io.d.ts

```diff
@@ -1,96 +1,823 @@
+		 * @param fn The function to remove, or null if we want to remove all functions
+		 * @return This Socket
+		 */
+		removeListener( event: string, fn?: Function ): Socket;
+		
+		/**
+		 * Removes all event listeners on this object
+		 * @return This Socket
+		 */
+		removeAllListeners(): Socket;
+		
+		/**
+		 * Sets the maximum number of listeners this instance can have
+		 * @param n The max number of listeners we can add to this emitter
+		 * @return This Socket
+		 */
+		setMaxListeners( n: number ): Socket;
+		
+		/**
+		 * Returns all the callbacks for a particular event
+		 * @param event The event that we're looking for the callbacks of
+		 * @return An array of callback Functions, or an empty array if we don't have any
+		 */
+		listeners( event: string ):Function[];
+	}
+	
+	/**
+	 * The interface used when dealing with rooms etc
+	 */
+	interface Adapter extends NodeJS.EventEmitter {
+		
+		/**
+		 * The namespace that this adapter is for
+		 */
+		nsp: Namespace;
+		
+		/**
+		 * A dictionary of all the rooms that we have in this namespace, each room
+		 * a dictionary of all the sockets currently in that room
+		 */
+		rooms: {[room: string]: {[id: string]: boolean }};
+		
+		/**
+		 * A dictionary of all the socket ids that we're dealing with, and all
+		 * the rooms that the socket is currently in
+		 */
+		sids: {[id: string]: {[room: string]: boolean}};
+		
+		/**
+		 * Adds a socket to a room. If the room doesn't exist, it's created
+		 * @param id The ID of the socket to add
+		 * @param room The name of the room to add the socket to
+		 * @param callback An optional callback to call when the socket has been 
+		 * added. It should take an optional parameter, error, if there was a problem
+		 */
+		add( id: string, room: string, callback?: ( err?: any ) => void ): void;
+		
+		/**
+		 * Removes a socket from a room. If there are no more sockets in the room,
+		 * the room is deleted
+		 * @param id The ID of the socket that we're removing
+		 * @param room The name of the room to remove the socket from
+		 * @param callback An optional callback to call when the socket has been
+		 * removed. It should take on optional parameter, error, if there was a problem
+		 */
+		del( id: string, room: string, callback?: ( err?: any ) => void ): void;
+		
+		/**
+		 * Removes a socket from all the rooms that it's joined
+		 * @param id The ID of the socket that we're removing
+		 */
+		delAll( id: string ):void;
+		
+		/**
+		 * Broadcasts a packet
+		 * @param packet The packet to broadcast
+		 * @param opts Any options to send along:
+		 * 	- rooms: An optional list of rooms to broadcast to. If empty, the packet is broadcast to all sockets
+		 * 	- except: A list of Socket IDs to exclude
+		 * 	- flags: Any flags that we want to send along ('json', 'volatile', 'broadcast')
+		 */
+		broadcast( packet: any, opts: { rooms?: string[], except?: string[], flags?: {[flag: string]: boolean} } ):void;
```

Comment:
you should use `broadcast( packet: any, opts: { rooms?: string[]; except?: string[]; flags?: {[flag: string]: boolean;}; } ):void;`. not comma.